notion-server/src/ethereum/mod.rs

```diff
@@ -663,4 +727,66 @@ mod tests {
             "value saved should match initial args passed in"
         );
     }
+
+    fn produce_block(wasm_code: &[u8], initial_state: &str, data: &str, expected_end_state: &str) {
+        let mut simulation = Simulation::new();
+
+        let execution_environment = args::ExecutionEnvironment {
+            base64_encoded_wasm_code: base64::encode(wasm_code),
+            base64_encoded_initial_state: base64::encode(&Vec::from_hex(initial_state).unwrap()),
+        };
+        assert_eq!(
+            0,
+            simulation
+                .create_execution_environment(args::CreateExecutionEnvironment {
+                    execution_environment,
+                })
+                .unwrap()
+        );
+
+        assert_eq!(0, simulation.create_shard_chain(args::CreateShardChain {}));
+
+        let shard_block = args::ShardBlock {
+            transactions: vec![args::ShardTransaction {
+                base64_encoded_data: base64::encode(&Vec::from_hex(data).unwrap()),
+                ee_index: 0,
+            }],
+        };
+        assert_eq!(
+            0,
+            simulation
+                .create_shard_block(args::CreateShardBlock {
+                    shard_chain_index: 0,
+                    shard_block,
+                })
+                .unwrap()
+        );
+
+        let post_state = simulation.shard_chains[0]
+            .execution_environment_state
+            .get(&EeIndex(0))
+            .unwrap();
+        let expected_state: [u8; 32] = FromHex::from_hex(expected_end_state).unwrap();
```

Comment:
Are we using hex so that it's consistent with the existing test cases in the `scout` repo?  Would it make more sense for us to use `base64` everywhere for consistency?

Reply:
I think for now it might make sense to stick with hex here to avoid having to convert the Scout tests.
I moved the `hex` crate to the `[dev-dependencies]` section, so it at least won't create overhead for release builds.

Reply:
works for me